src/connector/src/source/reader/reader.rs

```diff
@@ -105,6 +105,11 @@ impl SourceReader {
         }
     }
 
+    /// Postgres and Oracle connectors need to commit the offset to upstream.
+    pub fn need_commit_offset_to_upstream(&self) -> bool {
+        matches!(&self.config, ConnectorProperties::PostgresCdc(_))
```

Comment:
How about mysql? I thought all dbz-based cdc needs to commit offset.

Reply:
Do you mean that `recordCommitter.markProcessed()` and `recordCommitter.markBatchFinished` are no-ops for other dbz-based cdc source?

Reply:
No. only a step in `recordCommitter.markBatchFinished` is no-op for cdc sources that are not pg and oracle.
```java
/**
     * Commits the given offset with the source database. Used by some connectors
     * like Postgres and Oracle to indicate how far the source TX log can be
     * discarded.
     */
    default void commitOffset(Map<String, ?> partition, Map<String, ?> offset) {
    }
```
I got your point. To be cautious, we should call those methods for all cdc connectors.